src/decoder/stream.rs

```diff
@@ -964,34 +981,53 @@ impl StreamingDecoder {
                 }
 
                 self.state = Some(State::ImageData(chunk::fdAT));
-                Ok(Decoded::PartialChunk(chunk::fdAT))
+                Ok(Decoded::Nothing)
             }
         }
     }
 
     fn parse_chunk(&mut self, type_str: ChunkType) -> Result<Decoded, DecodingError> {
-        self.state = Some(State::new_u32(U32ValueKind::Crc(type_str)));
         let mut parse_result = match type_str {
+            // Critical non-data chunks.
             IHDR => self.parse_ihdr(),
-            chunk::sBIT => self.parse_sbit(),
             chunk::PLTE => self.parse_plte(),
+            chunk::IEND => Ok(()), // TODO: Check chunk size.
+
+            // Data chunks handled separately.
+            chunk::IDAT => Ok(()),
+            chunk::fdAT => Ok(()),
+
+            // Recognized bounded-size ancillary chunks.
+            chunk::sBIT => self.parse_sbit(),
             chunk::tRNS => self.parse_trns(),
             chunk::pHYs => self.parse_phys(),
             chunk::gAMA => self.parse_gama(),
             chunk::acTL => self.parse_actl(),
             chunk::fcTL => self.parse_fctl(),
             chunk::cHRM => self.parse_chrm(),
             chunk::sRGB => self.parse_srgb(),
-            chunk::cICP => Ok(self.parse_cicp()),
-            chunk::mDCV => Ok(self.parse_mdcv()),
-            chunk::cLLI => Ok(self.parse_clli()),
-            chunk::eXIf => Ok(self.parse_exif()),
-            chunk::bKGD => Ok(self.parse_bkgd()),
+            chunk::cICP => self.parse_cicp(),
+            chunk::mDCV => self.parse_mdcv(),
+            chunk::cLLI => self.parse_clli(),
+            chunk::bKGD => self.parse_bkgd(),
```

Comment:
I wonder if we should instead treat *all* parse errors in non-critical chunk, including their positions in the chunk stream, as `SkippedAncillaryChunk` as well. Well, not really blocking.